Core/Core/PageViewAnalytics/PageViewEventController.swift

```diff
@@ -144,20 +144,16 @@ public class PageViewEventController: NSObject {
     private func populatePlaceholderUrl(urlWithPlaceholders: String?, params: PageViewEventDictionary?) -> String? {
         guard let baseURL = LoginSession.mostRecent?.baseURL,
             let urlWithPlaceholders = urlWithPlaceholders
-            else { return nil }
+        else { return nil }
+        //  return url if it's already a full url
+        if let isFullyQualifiedUrl = URL(string: urlWithPlaceholders), isFullyQualifiedUrl.scheme == "http" || isFullyQualifiedUrl.scheme == "https" {
+            return urlWithPlaceholders
+        }
         var path = urlWithPlaceholders
         if let customPageViewPath = params?[PageViewEventController.Constants.customPageViewPath]?.description { path = customPageViewPath }
         if let paramterizedUrl = path.populatePathWithParams(params) {
             path = paramterizedUrl
         }
-        if (path.hasPrefix("/")) {
-            path = String(path.dropFirst())
```

Comment:
This is where the exception was. Not sure why it only happened in release. I couldn't repro in a playground.

```
2019-09-23 15:34:00.015821-0600 Student[4746:1046048] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[NSPathStore2 characterAtIndex:]: index (1) beyond bounds (1)'
```

`urlWithPlaceholders` was `"/"` when I caught the exception. Release mode must not like creating a String from an empty substring?

Reply:
Anywho, we don't need to drop the leading `/` because path is used in `baseURL.appendingPathComponent` and it works the same with leading slash or not.